http.go

```diff
@@ -88,6 +89,10 @@ func watchHandler(response http.ResponseWriter, req *http.Request, list *memberl
 
 	var jsonBytes []byte
 	pushUpdate := func() error {
+		// Enter critical section
+		state.RLock()
+		defer state.RUnlock()
```

Comment:
Indeed @bparli, it's probably not such a good idea to lock the state for so long here.